### PR TITLE
Maven flatten plugins for examples

### LINKED ISSUE #13512: Introduce Maven Flatten Plugin for project examples
### Is your feature request related to a problem?
There are too many modules in examples and we have to update example versions in every example `pom.xml` during project release, which is tedious to maintain.

### Describe the feature you would like.

Using `maven-flatten-plugin` in examples and replace version with variable in `pom.xml`.

### PR DESCRIPTION
**Pull request by Swastyy**
Fixes #13512 

Some points:
- The variable I used for the version of the parent directory that is `/examples` is revision as the revised version of the project.
- In the sub directories of `examples` Should I remove the plugin management lines? @TeslaCN 
- Let me know if there is anything that I am missing or doing incorrect.

### CODE REVIEW

**Comment by Swastyy** [comment]
@TeslaCN Is the pom files inside the sub directories of examples (meaning the third level of hierarchy) need to have changed to the version variable or they will remain as it is?

File: examples/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.apache.shardingsphere.example</groupId>
     <artifactId>shardingsphere-example</artifactId>
-    <version>5.0.1-SNAPSHOT</version>
+    <version>${revision}</version>
```
Comment (TeslaCN) [inline examples/pom.xml:25]: Hi @Swastyy 
Sorry for the delay. You may need to define a value for `revision` in properties.

File: examples/shardingsphere-proxy-example/pom.xml [hunk 2]
```diff
@@ -33,5 +33,38 @@
         <module>shardingsphere-proxy-boot-mybatis-example</module>
         <module>shardingsphere-proxy-hint-example</module>
     </modules>
+
+    <build>
+		<pluginManagement>
```
Comment (TeslaCN) [inline examples/shardingsphere-proxy-example/pom.xml:38]: Please fix the indent here and other.

File: examples/pom.xml (hunk 1 above)
Comment (Swastyy) [reply examples/pom.xml]: Yea, I have added that 👍

File: examples/shardingsphere-proxy-example/pom.xml (hunk 2 above)
Comment (Swastyy) [reply examples/shardingsphere-proxy-example/pom.xml]: In ide it looks correctly indented, I am not sure why it looks like here this way, I will correct the individual files indentation if it still happens.

Comment (Swastyy) [reply examples/shardingsphere-proxy-example/pom.xml]: @TeslaCN The CI test are failing with error as they can't find the version of parent although I have added it correctly, and there are no dependencies in those pom files, I cannot understand what is missing, please help.

File: examples/pom.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.apache.shardingsphere.example</groupId>
     <artifactId>shardingsphere-example</artifactId>
-    <version>5.0.1-SNAPSHOT</version>
+    <version>${revision}${sha1}${changelist}</version>
```
Comment (TeslaCN) [inline examples/pom.xml:25]:
```suggestion
    <version>${revision}</version>
```

File: examples/pom.xml [hunk 4]
```diff
@@ -35,6 +35,9 @@
     </modules>
     
     <properties>
+        <revision>5.0.1</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <sha1/>
```
Comment (TeslaCN) [inline examples/pom.xml:40]:
```suggestion
        <revision>5.0.1-SNAPSHOT</revision>
```

**Review by TeslaCN — CHANGES_REQUESTED** [review]
You may refer to this PR https://github.com/apache/shardingsphere-elasticjob/pull/1672

**Comment by TeslaCN** [comment]
@Swastyy Could you figure out why apache/shardingsphere-elasticjob#1672 worked?

File: examples/shardingsphere-proxy-example/pom.xml (hunk 2 above)
Comment (TeslaCN) [reply examples/shardingsphere-proxy-example/pom.xml]: Your indent are `tab`, replace them with `space`.

**Comment by Swastyy** [comment]
> @TeslaCN Is the pom files inside the sub directories of examples (meaning the third level of hierarchy) need to have changed to the version variable or they will remain as it is?

Hey @TeslaCN I could see this as the problem, we are having so many levels and upto which one the parent version should be changed to variable, now in https://github.com/apache/shardingsphere-elasticjob/pull/1672 there is only two levels so it was handled easily but for more than 3 levels should I change every pom in the depth to revision?

**Comment by TeslaCN** [comment]
> Hey @TeslaCN I could see this as the problem, we are having so many levels and upto which one the parent version should be changed to variable, now in [apache/shardingsphere-elasticjob#1672](https://github.com/apache/shardingsphere-elasticjob/pull/1672) there is only two levels so it was handled easily but for more than 3 levels should I change every pom in the depth to revision?

All versions of examples' pom should be changed to placeholder. What we expect is changing one place can change all examples' version.

**Comment by Swastyy** [comment]
> > Hey @TeslaCN I could see this as the problem, we are having so many levels and upto which one the parent version should be changed to variable, now in [apache/shardingsphere-elasticjob#1672](https://github.com/apache/shardingsphere-elasticjob/pull/1672) there is only two levels so it was handled easily but for more than 3 levels should I change every pom in the depth to revision?
> 
> All versions of examples' pom should be changed to placeholder. What we expect is changing one place can change all examples' version.

I have completed it. Please review it once.

File: examples/example-core/pom.xml [hunk 5]
```diff
@@ -36,4 +36,37 @@
         <module>example-spring-mybatis</module>
         <module>example-spring-jpa</module>
     </modules>
+
+    <build>
+        <pluginManagement>
+		    <plugins>
```
Comment (TeslaCN) [inline examples/example-core/pom.xml:42]: The indents here and below mixed `tab` and `space` and they are still incorrect. How about just removing the `lifecycle-mapping`?

Comment (Swastyy) [reply examples/example-core/pom.xml:42]: Yea, there is no specific use of it(a comment also mentions), I'll remove the `lifecycleMapping`. Also @TeslaCN there are too many changed files in this PR can I do it another PR?

Comment (TeslaCN) [reply examples/example-core/pom.xml:42]: I prefer doing it in this PR. The examples updated in https://github.com/apache/shardingsphere/pull/13931.
You may need to update your branch.

**Comment by Swastyy** [comment]
Hi @TeslaCN I have corrected the indents and updated the poms after #13931

**Review by TeslaCN — APPROVED** [review]

**Comment by TeslaCN** [comment]
@Swastyy Thank you!

**Pull request merged**